### PR TITLE
feat: scripts created to delete duplicates having same solId and userId from project collection

### PR DESCRIPTION
**Pull request by borkarsaish65**
Delete duplicate records having same userId and solutionId to enforce compound key 
 and data integrity

### CODE REVIEW

File: migrations/deleteDuplicateRecordsAcrossCollections/deleteDuplicateRecords.js [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+      },
+    ];
+    let duplicateArray = await db
+      .collection("projects")
+      .aggregate(pipeline)
+      .toArray();
+
+    let successfullyDeletedRecords = [];
+    let failedToDeletedRecords = [];
+    for (let duplicateRecord of duplicateArray) {
```
Comment (ankit1shahu) [inline migrations/deleteDuplicateRecordsAcrossCollections/deleteDuplicateRecords.js]: create chunk and process each chunk for reference you can check another migration in same repo @borkarsaish65

File: migrations/deleteDuplicateRecordsAcrossCollections/deleteDuplicateRecords.js [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+          if (duplicateArray[i].status == "submitted") {
+            completedProjects.push(duplicateArray[i]);
+          } else if (duplicateArray[i].status == "started") {
+            inCompleteProjects.push(duplicateArray[i]);
+          }else if(duplicateArray[i].status == 'inProgress'){
+            inProgressProjects.push(duplicateArray[i])
+          }
+        }
+
+        completedProjects.sort(sortByCreatedAtDescending);
```
Comment (ankit1shahu) [inline migrations/deleteDuplicateRecordsAcrossCollections/deleteDuplicateRecords.js]: dont check only createdAt if project has duplicate then check which one is in started and which one is in inprogress then delete the started One keep the inprogress on @borkarsaish65

File: migrations/deleteDuplicateRecordsAcrossCollections/findDuplicateRecords.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+let mongoUrl = process.env.MONGODB_URL;
+let dbName = mongoUrl.split("/").pop();
+let url = mongoUrl.split(dbName)[0];
+var MongoClient = require("mongodb").MongoClient;
+
+var fs = require("fs");
+
+(async () => {
+  let connection = await MongoClient.connect(url, { useNewUrlParser: true });
+
```
Comment (ankit1shahu) [inline migrations/deleteDuplicateRecordsAcrossCollections/findDuplicateRecords.js]: we dont need this @borkarsaish65

File: migrations/deleteDuplicateRecordsAcrossCollections/deleteDuplicateRecords.js [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+            }
+
+      } catch (e) {
+        console.log(e);
+        continue;
+      }
+    }
+
+    fs.writeFileSync('successfully_deleted_duplicated_records' + generateUUId()+'.js',JSON.stringify(successfullyDeletedRecords))
+    fs.writeFileSync('failed_to_deleted_duplicated_records' + generateUUId()+'.js',JSON.stringify(failedToDeletedRecords))
```
Comment (ankit1shahu) [inline migrations/deleteDuplicateRecordsAcrossCollections/deleteDuplicateRecords.js:160]: you are creating one file with record which will have ids you can use that for validadation

**Review by ankit1shahu — CHANGES_REQUESTED** [review]

File: migrations/deleteDuplicateRecordsAcrossCollections/deleteDuplicateRecords.js (hunk 1 above)
Comment (borkarsaish65) [reply migrations/deleteDuplicateRecordsAcrossCollections/deleteDuplicateRecords.js]: Done this now

File: migrations/deleteDuplicateRecordsAcrossCollections/deleteDuplicateRecords.js (hunk 2 above)
Comment (borkarsaish65) [reply migrations/deleteDuplicateRecordsAcrossCollections/deleteDuplicateRecords.js]: I have used this exact logic. Please check below code. Here I am just sorting all the records in descending order.

File: migrations/deleteDuplicateRecordsAcrossCollections/findDuplicateRecords.js (hunk 3 above)
Comment (borkarsaish65) [reply migrations/deleteDuplicateRecordsAcrossCollections/findDuplicateRecords.js]: I tried removing this , it gives an error. All other scripts also have the let connection declaration.

File: migrations/deleteDuplicateRecordsAcrossCollections/deleteDuplicateRecords.js (hunk 4 above)
Comment (borkarsaish65) [reply migrations/deleteDuplicateRecordsAcrossCollections/deleteDuplicateRecords.js:160]: Ok Can do that.

File: migrations/deleteDuplicateRecordsAcrossCollections/Readme.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+Run the script to delete duplicate projects.
+
+    node deleteDuplicateRecords.js
+
+#### Validation 
+
+After the script has been executed in the staging environment, we can validate the data by utilizing the findDuplicateRecords.js script
+
+Run the script to delete duplicate projects.
```
Comment (ankit1shahu) [inline migrations/deleteDuplicateRecordsAcrossCollections/Readme.md]: This is not needed you can give db find API curl with payload @borkarsaish65

File: migrations/deleteDuplicateRecordsAcrossCollections/deleteDuplicateRecords.js [hunk 6]
```diff
@@ -0,0 +1,177 @@
...
+          duplicateArray: "$docs",
+        },
+      },
+    ];
+    let duplicateArray = await db
+      .collection("projects")
+      .aggregate(pipeline)
+      .toArray();
+
+    let duplicateProjectsChunk = _.chunk(duplicateArray,10);
```
Comment (ankit1shahu) [inline migrations/deleteDuplicateRecordsAcrossCollections/deleteDuplicateRecords.js]: make it 100 @borkarsaish65

**Review by ankit1shahu — CHANGES_REQUESTED** [review]

File: migrations/deleteDuplicateRecordsAcrossCollections/deleteDuplicateRecords.js (hunk 6 above)
Comment (borkarsaish65) [reply migrations/deleteDuplicateRecordsAcrossCollections/deleteDuplicateRecords.js]: Done

File: migrations/deleteDuplicateRecordsAcrossCollections/Readme.md (hunk 5 above)
Comment (borkarsaish65) [reply migrations/deleteDuplicateRecordsAcrossCollections/Readme.md]: Changed now

**Pull request closed without merging**